nncf/torch/dynamic_graph/graph.py

```diff
@@ -109,9 +124,14 @@ def __init__(self,
             DefaultTensorMetaComparator()]
         self.input_matcher = input_matcher if input_matcher else DefaultInputsMatcher()
 
-    def __eq__(self, other: 'OperationExecutionContext'):
-        return (self.op_address == other.op_address) and \
-               self.input_matcher(self.tensor_metas, other.tensor_metas, self.tm_comparators)
+    def __eq__(self, other):
+        return self.op_address == other.op_address and Counter(self.tensor_metas) == Counter(other.tensor_metas)
+
+    def matches_saved_inputs_from(self, other: 'OperationExecutionContext'):
+        # WARNING: not commutative
+        return self.op_address == other.op_address and self.input_matcher(other.tensor_metas,
+                                                                          self.tensor_metas,
+                                                                          self.tm_comparators)
```

Comment:
In which situations commutative comparison is used?
Is it theoretically possible that inputs, let's say for the Add and Concat modules, would come in the different order from time to time?

Reply:
Commutative comparison is used for comparing `DynamicGraph` instances which hold the `OperationExecutionContext`.